### PR TITLE
Add 6 vscode extensions

### PR DESCRIPTION
**Pull request by jackyliu16**
###### Description of changes
adding 5 VS Code extension about translate.

###### Things done
 
I have only test the full function of code-translate, but others haven't been fully tested, which I just add it by the way.
The things that I can promise is all extension could be build successes, and seems in the extension page without any kind of error log.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by drupol — CHANGES_REQUESTED** [review]
Please add metadata. Have a look at other extension for some inspiration!

**Review by drupol — CHANGES_REQUESTED** [review]
There are missing metadata and the descriptions must be updated.

See this comment and PR, it might help: https://github.com/NixOS/nixpkgs/pull/223717#discussion_r1152615779

**Review by drupol — CHANGES_REQUESTED** [review]
Please complete meta information:

- Start description as "A Visual Studio Code extension...."
- Consider adding yourself as maintainer.
- Use English language

**Comment by jackyliu16** [comment]
The Chinese use is because it has been use by the extension as a description, I'm already trying to reduce the use of Chinese. If it still needs to decrease, it may be necessary to use translation, which I guess it's not good for Chinese use, bc seems the extension is made by them, thus it's using Chinese for description.

**Comment by jackyliu16** [comment]
And I have a little confused about why we should repeat the package is a kind of vscode-extension, which i though when the reader has seemed the packages has a kind of subtitle such as `vscode-extensions.vscodevim.vim` and then will understand the packages is a kind of vscode extension is't it?

**Comment by drupol** [comment]
Maybe ask @superherointj ?

**Comment by superherointj** [comment]
> Maybe ask @superherointj ?

Honestly, I have no opinion.
Have a good day.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -412,6 +412,22 @@ let
...
       };
 
+      ban.spellright = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          publisher = "ban";
+          name = "spellright";
+          version = "3.0.112";
+          sha256 = "sha256-79Yg4I0OkfG7PaDYnTA8HK8jrSxre4FGriq0Baiq7wA=";
+        };
+        meta = {
```
Comment (drupol) [inline pkgs/applications/editors/vscode/extensions/default.nix:422]: Missing description

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -653,6 +669,31 @@ let
...
 
+      w88975.code-translate = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "code-translate";
+          publisher = "w88975";
+          version = "1.0.20";
+          sha256 = "sha256-blqLK7S+RmEoyr9zktS5/SNC0GeSXnNpbhltyajoAfw=";
+        };
+        meta = {
+          description  = "A Visual Studio Code purely hover translation extension";
```
Comment (drupol) [inline pkgs/applications/editors/vscode/extensions/default.nix:680]: A Visual Studio Code extension ...

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -1121,6 +1162,23 @@ let
...
 
+      ExiaHuang.dictionary = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          publisher = "ExiaHuang";
+          name = "dictionary";
+          version = "0.0.2";
+          sha256 = "sha256-caNcbDTB/F2mdlGpfIfJv13lzY5Wwj7p7r8dAte9+3A=";
+        };
+        meta = {
+          description   = "A vscode extension of using chinese-english dictonary in right-click menu";
```
Comment (drupol) [inline pkgs/applications/editors/vscode/extensions/default.nix:1173]: Please be consistent with other extension.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 4]
```diff
@@ -1248,6 +1306,24 @@ let
...
 
+      funkyremi.vscode-google-translate = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          publisher = "funkyremi";
+          name = "vscode-google-translate";
+          version = "1.4.13";
+          sha256 = "sha256-9Vo6lwqD1eE3zY0Gi9ME/6lPwmwuJ3Iq9StHPvncnM4=";
+        };
+        meta = {
+          description   = "Translate text right in your code";
```
Comment (drupol) [inline pkgs/applications/editors/vscode/extensions/default.nix:1317]: Consistency in description

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 5]
```diff
@@ -1502,6 +1578,27 @@ let
...
 
+      intellsmi.comment-translate = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          publisher = "intellsmi";
+          name = "comment-translate";
+          version = "2.2.4";
+          sha256 = "sha256-g6mlScxv8opZuqgWtTJ3k0Yo7W7WzIkwB+8lWf6cMiU=";
+        };
+        meta = {
+          description = "A vscode-extension showing the translateion of the word you mouse hover";
```
Comment (drupol) [inline pkgs/applications/editors/vscode/extensions/default.nix:1589]: Consistency

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 6]
```diff
@@ -3227,6 +3324,24 @@ let
...
 
+      yzhang.dictionary-completion = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          publisher = "yzhang";
+          name = "dictionary-completion";
+          version = "1.2.2";
+          sha256 = "sha256-dpJcJARRKzRNHfXs/qknud8OQ8xIyeaVnt/EcDq0k4E=";
+        };
+        meta = {
+          description = "Word Completion";
```
Comment (drupol) [inline pkgs/applications/editors/vscode/extensions/default.nix:3335]: Consistency

**Review by drupol — CHANGES_REQUESTED** [review]
The descriptions for these five extensions vary significantly in their structure.

It would be helpful if you could maintain a consistent format when writing these descriptions.

**Comment by jackyliu16** [comment]
EMM, I thought the sort of meta is unnecessary

**Comment by jackyliu16** [comment]
@superherointj Is it enough to reach this level?

**Review by drupol — CHANGES_REQUESTED** [review]
I'm sorry, no.

Could you please try to be more consistent in the way you write description for the **6 extensions** (not 5) you're trying to add ?

E.g.: try to write `vscode` in the same way everywhere. In one of them you write `vscode`, in another one `Visual Studio Code`... Do you follow me? These descriptions will end up on https://search.nixos.org, it is important to be consistent and regular.

Here are the 6 descriptions, none of them match a common pattern:

- `description = "Multilingual, Offline and Lightweight Spellchecker for Visual Studio Code";`
- `description  = "A Visual Studio Code purely hover translation extension";`
- `description   = "A vscode extension of using chinese-english dictonary in right-click menu";`
- `description   = "A VSCode Extension to using google translateion to helping you quickly translate text right in your code rocket";`
- `description = "A vscode-extension showing the translateion of the word you mouse hover";`
- `description = "VSCode Dictionary Completion Extension";`

Can you please rephrase correctly the `description`  attributes as suggested in [a previous comment](https://github.com/NixOS/nixpkgs/pull/227581#issuecomment-1518685870) and make sure code alignment is consistent (`the equals signs`), throughout your changes in this PR.

**Comment by jackyliu16** [comment]
> Please complete meta information:
> 
>     * Start description as "A Visual Studio Code extension...."
> 
>     * Consider adding yourself as maintainer.
> 
>     * Use English language

Matching request, @drupol could you provide the specify Community Guidelines or a kind of RFC about the vscode-extension should start by `A Visual Studio Code extension` ? This doesn't look like a community norms  , because I found there are only 35 descriptions of compliance throughout the document.

**Comment by jackyliu16** [comment]
`make sure code alignment is consistent (the equals signs),`, i not sure if you means the equals signs should be a line or others, if it's what your means, seems it has been reject by `nixpkgs-fmt`

**Comment by drupol** [comment]
> Matching request, @drupol could you provide the specify Community Guidelines or a kind of RFC about the vscode-extension should start by `A Visual Studio Code extension` ? This doesn't look like a community norms , because I found there are only 35 descriptions of compliance throughout the document.

While there may not be any official community guidelines or RFC specifically about the naming conventions for Visual Studio Code extensions, it's important to maintain consistency and follow the general best practices for extension descriptions.

Considering your past experiences and to ensure a consistent style across different extensions, it's reasonable to suggest that the extension's description should start with "A Visual Studio Code extension". I asked you to fix your PR since it was asked to me when I submitted the same kind of PR (see my previous comments about it).

**Review by drupol — APPROVED** [review]
Beside the PR title (6 instead of 5), LGTM.

**Comment by jackyliu16** [comment]
:) 6 is BC i forgot the first commit haven't been merged

**Review by drupol — APPROVED** [review]
LGTM

**Pull request merged**